sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/subquery.scala

```diff
@@ -106,12 +106,20 @@ object RewritePredicateSubquery extends Rule[LogicalPlan] with PredicateHelper {
 
       // Filter the plan by applying left semi and left anti joins.
       withSubquery.foldLeft(newFilter) {
-        case (p, Exists(sub, conditions, _)) =>
-          val (joinCond, outerPlan) = rewriteExistentialExpr(conditions, p)
-          buildJoin(outerPlan, sub, LeftSemi, joinCond)
-        case (p, Not(Exists(sub, conditions, _))) =>
-          val (joinCond, outerPlan) = rewriteExistentialExpr(conditions, p)
-          buildJoin(outerPlan, sub, LeftAnti, joinCond)
+        case (p, exists @ Exists(sub, conditions, _)) =>
```

Comment:
can we change the beginning instead?
```
val (withSubquery, withoutSubquery) = splitConjunctivePredicates(condition).partition { cond =>
  SubqueryExpression.hasInOrExistsSubquery(cond) && !isNonCorrelatedExists
}
```

Reply:
or we change `hasInOrExistsSubquery` to `hasInOrCorrelatedExistsSubquery `

Reply:
> hasInOrCorrelatedExistsSubquery

Done